lib/examples/did-test-suite.rs

```diff
@@ -0,0 +1,629 @@
+            did_document_stream: representation,
+            did_resolution_metadata: res_meta,
+            did_document_metadata: doc_meta,
+        };
+        let execution = ResolverExecution {
+            function: ResolverFunction::ResolveRepresentation,
+            input,
+            output,
+        };
+        self.add_execution(execution, error_opt, deactivated_opt);
+    }
+
+    async fn dereference(
+        &mut self,
+        resolver: &dyn DIDResolver,
+        did_url: &str,
+        options: &DereferencingInputMetadata,
+    ) {
+        let (deref_meta, content, content_meta) = dereference(resolver, did_url, options).await;
+        let input = ExecutionInput::Dereference {
+            did_url: did_url.to_string(),
+            dereference_options: options.to_owned(),
+        };
+        let error_opt = deref_meta.error.clone();
+        let deactivated_opt = if let ContentMetadata::DIDDocument(ref did_doc_meta) = content_meta {
+            did_doc_meta.deactivated
+        } else {
+            None
+        };
+        let content_stream = match content {
+            Content::DIDDocument(doc) => serde_json::to_string(&doc).unwrap(),
+            Content::URL(url) => String::from(url),
+            Content::Object(resource) => serde_json::to_string(&resource).unwrap(),
+            Content::Data(vec) => String::from_utf8(vec).unwrap(),
+            Content::Null => "".to_string(),
+        };
+        let output = ExecutionOutput::Dereference {
+            dereferencing_metadata: deref_meta,
+            content_stream,
+            content_metadata: content_meta,
+        };
+        let execution = ResolverExecution {
+            function: ResolverFunction::Dereference,
+            input,
+            output,
+        };
+        self.add_execution(execution, error_opt, deactivated_opt);
+    }
+
+    fn add_execution(
+        &mut self,
+        execution: ResolverExecution,
+        error_opt: Option<String>,
+        deactivated_opt: Option<bool>,
+    ) {
+        let i = self.executions.len();
+        self.executions.push(execution);
+        let outcome = ResolverOutcome::from_error_or_deactivated(error_opt, deactivated_opt);
+        self.expected_outcomes.entry(outcome).or_default().push(i);
+    }
+}
+
+async fn report_resolver_key() {
+    let mut report = DIDResolverImplementation {
+        did_method: "did:key".to_string(),
+        implementation: "https://github.com/spruceid/ssi/tree/main/did-key".to_string(),
+        implementer: "Spruce Systems, Inc.".to_string(),
+        expected_outcomes: HashMap::new(),
+        executions: Vec::new(),
+    };
+
+    for did in vec![
+        "did:key:z6MkpTHR8VNsBxYAAWHut2Geadd9jSwuBV8xRoAnwWsdvktH", // Ed25519
+        "did:key:zQ3shokFTS3brHcDQrn82RUDfCZESWL1ZdCEJwekUDPQiYBme", // Secp256k1
+        "did:key:zDnaerDaTF5BXEavCrfRZEk316dpbLsfPDZ3WJ5hRTPFU2169", // Secp256r1
+        "did:key;invalid",
```

Comment:
Is the `;` voluntary?

Reply:
Yes, intended make the resolve fail with "invalidDid" error

Reply:
Ok :+1: was wondering if the goal was to have an invalid key

Reply:
Added a comment to note it in ce164b0e5fd20d6dab99b8408c387252089d243d